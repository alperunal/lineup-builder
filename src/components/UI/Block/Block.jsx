import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';

import './Block.scss';

const Block = ({ title, customClass, children }) => (
  <div className={cn('block', customClass)}>
    {title ? <div className="heading">{title}</div> : null}
    <div className="content">{children}</div>
  </div>
);

Block.propTypes = {
  title: PropTypes.string,
  customClass: PropTypes.string,
  children: PropTypes.node.isRequired,
};

Block.defaultProps = {
  title: '',
  customClass: '',
};

export default Block;
