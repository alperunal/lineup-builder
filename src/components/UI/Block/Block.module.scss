.block {
    background-color: var(--color-block-background);
    position: relative;
    padding: 0;
    border-radius: 6px;
    border-width: 1px;
    border-style: solid;
    border-color: var(--color-border);
    box-shadow: 0 0 40px 0 rgba(94,92,154,.06);
    margin-top: 0;
    margin-bottom: 0;

    .heading {
        border-bottom-width: 1px;
        border-bottom-style: solid;
        border-bottom-color: var(--color-border);
        padding: 20px 20px;
        margin-bottom: 0;
        padding-bottom: 20px;
        font-size: 17px;
        color: var(--color-text);
        font-weight: 600;
    
    }

    .content {
        padding: 20px;
    }
}