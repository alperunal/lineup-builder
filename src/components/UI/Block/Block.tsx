import React from 'react';
import cn from 'classnames';
import './Block.module.scss';

type Props = {
    title?: string,
    customClass?: string,
    children: React.ReactNode,
};

const Block = ({
    title, customClass, children,
}: Props): JSX.Element => (
    <div
        className={
            cn(
                'block',
                customClass,
            )
        }
    >
        {title ? (
            <div className="heading">
                {title}
            </div>
        ) : null}
        <div className="content">
            {children}
        </div>
    </div>
);

Block.defaultProps = {
    customClass: '',
};

export default Block;
