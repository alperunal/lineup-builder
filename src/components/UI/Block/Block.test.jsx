import React from 'react';
import { shallow } from 'enzyme';
import Block from './Block';

it('should render Block correctly', () => {
  const wrapper = shallow(
    <Block title="Test">
      <div>Some children components</div>
      <span>Some childrens x2</span>
    </Block>,
  );
  expect(wrapper).toMatchSnapshot();
});
