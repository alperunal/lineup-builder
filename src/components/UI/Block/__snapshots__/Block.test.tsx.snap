// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Block correctly 1`] = `ShallowWrapper {}`;
